### ISSUE TITLE
s2i steps default to git ref "master", but not all git repositories have a master branch

### ISSUE THREAD

**Issue by jmtd** [cekit/behave-test-steps#49]
In the absence of an explicit ref in the step "Given s2i build...", i.e., `Given s2i build https://...somerepo.../` with no `using branchname` suffix, the S2I steps will default to try to check out a branch named `master`, which may not exist in the repository.

Instead I think it should use the repository's default branch name.

We hit this ([side issue](https://github.com/jboss-container-images/openjdk/pull/405#issuecomment-1799779819)) by introducing a new repository for cekit/behave tests, https://github.com/jboss-container-images/openjdk-test-applications, which has the default branch name `main`. This is the new default, I think, for both Git and Github. (At the time of writing this, I'm about to push a `master` branch to that repository as a temporary mitigation).

The culprits are
https://github.com/cekit/behave-test-steps/blob/e92a2391d866ee74a733cbc8bbb203dbbe452655/steps/s2i_steps.py#L14
and
https://github.com/cekit/behave-test-steps/blob/e92a2391d866ee74a733cbc8bbb203dbbe452655/steps/s2i_steps.py#L59
and possibly a few other places.

**Comment by rnc** [comment]
@jmtd You just updated `failing_s2i_build` in https://github.com/cekit/behave-test-steps/pull/74 to remove the explicit tag option but given the s2i clones specifying the reference (i.e. `master` or `main` etc) then unless some work is done to first establish the remote default branch (e.g. `git ls-remote --symref git@github.com:rnc/behave-test-steps.git HEAD | head -1 | awk '{print $2}' | sed 's/refs\/heads\///'`) then I don't think we can just remove `master`. Potentially removing the default and _if empty_, then use default branch name might work?

**Comment by jmtd** [comment]
FWIW, I am incorrect in my statement above that Git's default has changed, it's _still_ master.

**Comment by rnc** [comment]
> FWIW, I am incorrect in my statement above that Git's default has changed, it's _still_ master.

Its now configurable (https://github.blog/open-source/git/highlights-from-git-2-28/#introducing-init-defaultbranch) and gitlab, github, bitbucket have already made the changes.

**Comment by jmtd** [comment]
> Its now configurable

Yeah… I've hit a problem in other software where my global preference (`git config --global…`) broke an implicit assumption that the value is `master`… this will be a problem all over the place eventually
